main.go

```diff
@@ -88,14 +89,16 @@ func runMain() error {
 			}
 			// Start with an empty previous frame for a new recording.
 			prevFrame = new(lepton3.Frame)
-		} else if recordingCount == 0 && writer != nil {
+		} else if recordingCount == 0 && writer != nil ||
+				recordingFrameLen > conf.MaxLen {
```

Comment:
As per earlier comments, the maximum recorded frames should be calculated from the maximum seconds in the config multiplied by framesHz.